content/academy/data_collection_basics/api_scraping.md

```diff
@@ -122,9 +127,61 @@ const HEADERS = {
 }
 ```
 
-However, a much better option is to use either a custom implementation of generating random headers for each request, or to use a package such as <a href="https://www.npmjs.com/package/header-generator">`header-generator`</a> to automatically do this.
+However, a much better option is to use either a custom implementation of generating random headers for each request, or to use a package such as [`got-scraping`](https://www.npmjs.com/package/got-scraping) to automatically do this.
+
+With `got-scraping`, generating request-specific headers can be done right within a request with `headerGeneratorOptions`. Specific headers can also be set with the "`headers`" option:
 
-For our SoundCloud example, testing the endpoint from the previous section in a tool like <a href="https://www.postman.com/">Postman</a> works perfectly, and returns the data we want; however, when the "`client_id`" parameter is removed, we receive a `401 Unauthorized` error. Luckily, the Client ID is the same for every user, which means that it is not tied to a session or an IP address (this is based on our own observations and tests). The big downfall is that the token being used by SoundCloud changes every few weeks, so it shouldn't be hardcoded. **A very strange case indeed**.
+```JavaScript
+const response = await gotScraping({
+    url: 'https://example.com',
+    headerGeneratorOptions:{
+        browsers: [
+            {
+                name: 'chrome',
+                minVersion: 87,
+                maxVersion: 89
+            }
+        ],
+        devices: ['desktop'],
+        locales: ['de-DE', 'en-US'],
+        operatingSystems: ['windows', 'linux'],
+    },
+    headers: {
+        'some-header': 'Hello, Academy!'
+    }
+})
+```
+
+For our SoundCloud example, testing the endpoint from the previous section in a tool like [Postman](https://www.postman.com/) works perfectly, and returns the data we want; however, when the "`client_id`" parameter is removed, we receive a `401 Unauthorized` error. Luckily, the Client ID is the same for every user, which means that it is not tied to a session or an IP address (this is based on our own observations and tests). The big downfall is that the token being used by SoundCloud changes every few weeks, so it shouldn't be hardcoded. This case is actually quite common, and is not only seen with SoundCloud.
```

Comment:
The link should go to the Postman article. For two reasons. Keeping the user longer on our domain is good for SEO and we can explain the web scraping intricacies of postman better in our own article than just linking to the Postman docs.